specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/bastionHost.json

```diff
@@ -0,0 +1,440 @@
+                   "$ref":"./network.json#/parameters/ApiVersionParameter"
+                },
+                {  
+                   "$ref":"./network.json#/parameters/SubscriptionIdParameter"
+                }
+             ],
+             "responses":{  
+                "200":{  
+                    "description":"Success. The operation returns BastionHost associated with network.",
+                    "schema":{  
+                     "$ref":"#/definitions/BastionHost"
+                    }
+                 }
+             },
+             "x-ms-examples":{  
+                "Get Bastion Host for a given vnet":{  
+                   "$ref":"./examples/BastionHostListByVnet.json"
+                }
+             }
+          }
+       }
+    },
+    "definitions":{  
+       "BastionHostIPConfigurationPropertiesFormat":{  
+          "properties":{  
+             "subnet":{
+                "$ref":"./network.json#/definitions/SubResource",
+                "description":"Reference of the subnet resource. This field is a mandatory input."
+             },
+             "publicIPAddress": {
+                "$ref": "./network.json#/definitions/SubResource",
+                "description": "Reference of the PublicIP resource. This field is a mandatory input."
+             },
+             "provisioningState":{
+               "description":"The provisioning state of the resource.",
+               "$ref":"./network.json#/definitions/ProvisioningState"
+             },
+             "privateIPAllocationMethod": {
+               "type": "string",
+               "description": "PrivateIP allocation method.",
+               "enum": [
+                 "Static",
+                 "Dynamic"
+               ],
+               "x-ms-enum": {
+                 "name": "IPAllocationMethod",
+                 "modelAsString": true
+               }
+            }
+          },
+          "description":"Properties of IP configuration of an Bastion Host."
+       },
+       "BastionHostIPConfiguration":{  
+          "properties":{  
+             "properties":{  
+                "x-ms-client-flatten":true,
+                "$ref":"#/definitions/BastionHostIPConfigurationPropertiesFormat",
+               "description":"Represents the ip configuration associated with the resource."
+             },
+             "name":{  
+                "type":"string",
+                "readOnly": true,
+                "description":"Name of the resource that is unique within a resource group. This name can be used to access the resource."
+             },
+             "etag":{  
+                "type":"string",
+                "readOnly": true,
+                "description":"A unique read-only string that changes whenever the resource is updated."
+             },
+             "type": {
+               "readOnly": true,
+               "type": "string",
+               "description": "Ip configuration type."
+             },
+             "id": {
```

Comment:
Should be inherited from (differ by `"readOnly": true,`)
```
          "allOf":[  
             {  
                "$ref":"./network.json#/definitions/SubResource"
             }
          ],
```

If you need to override, I think you may omit allOf inheritance

Reply:
I have removed allof for this one. I think ID should be made readonly in subresource aswell. You can't really change id

Reply:
Actually i am adding subresource back and inheriting id from there. If its not readonly then its fine.